### PR TITLE
Refactoring profile view

### PR DESCRIPTION
**Pull request by Cheshiriks**
Removed coroutine scope

### CODE REVIEW

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt [hunk 1]
```diff
@@ -46,36 +43,38 @@ val profileSettingsCard: FC<SettingsProps> = FC { props ->
     val avatarWindowOpen = useWindowOpenness()
     val (settingsInputFields, setSettingsInputFields) = useState(SettingsInputFields())
     val (selectedAvatar, setSelectedAvatar) = useState<String?>(null)
+    val (avatar, setAvatar) = useState<File?>(null)
     val saveUser = useSaveUser(props, settingsInputFields, setSettingsInputFields)
 
+    val saveAvatar = useDeferredRequest {
+        avatar?.let {
+            val response = request(
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt:51]: is typed `post` method available here?

Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt]: Yes, it's work. But changed it

**Review by nulls — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt [hunk 2]
```diff
@@ -46,36 +43,38 @@ val profileSettingsCard: FC<SettingsProps> = FC { props ->
...
+                    owner = props.userInfo?.name
+                    this.type = AvatarType.USER
+                },
+                Headers().apply { append(CONTENT_LENGTH_CUSTOM, avatar.size.toString()) },
+                FormData().apply { set(FILE_PART_NAME, avatar) },
+                loadingHandler = ::noopLoadingHandler,
+                responseHandler = ::noopResponseHandler,
+            )
+            if (response.ok) {
+                window.location.reload()
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt:63]: Yes, the avatar is passed to the props and they won't update without a reload.

[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt:63]: Did we fix 401 btw?

[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/right/Profile.kt:63]: Not sure, I didn't fix it

**Pull request merged**